pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -341,6 +342,10 @@ private BrokerResponseNative handleSQLRequest(long requestId, String query, Json
       return BrokerResponseNative.NO_TABLE_RESULT;
     }
 
+    if (isDisableGroovy(realtimeTableConfig, offlineTableConfig)) {
```

Comment:
Only check the table config when that table is actually queried
```suggestion
    if (isDisableGroovy(offlineTableName != null ? offlineTableConfig: null, realtimeTableName != null ? realtimeTableConfig: null)) {
```

Reply:
Makes sense. Added.